lib/sycamore/sycamore/transforms/summarize.py

```diff
@@ -57,137 +58,375 @@ class LLMElementTextSummarizer(Summarizer):
+        Please combine these partial answers into a coherent single answer to the question "{{ question }}".
+        Some answers may not be particularly relevent, so don't pay them too much mind.
+        Answers:
+        {%- endif -%}
+        {%- for idx in elt.properties[batch_key] %}
+        {{ loop.index }}: {{ get_text(doc.elements[idx]) }}
+        {% endfor %}
+        """
+    ),
+    question="What is the summary of this data?",
+)
+
+
+class MultiStepDocumentSummarizer(Summarizer):
+    """
+    Summarizes a document by constructing a tree of summaries. Each leaf contains as many consecutive
+    elements as possible within the token limit, and each vertex of the tree contains as many sub-
+    summaries as possible within the token limit. e.g with max_tokens=10
 
-    @timetrace("SummText")
-    def _summarize_text_element(self, element: Element) -> Element:
-        prompt = _TextSummarizerGuidancePrompt()
+    .. code-block::
 
-        if element.text_representation:
-            response = self._llm.generate_old(prompt_kwargs={"prompt": prompt, "query": element.text_representation})
-            element.properties["summary"] = response
-        return element
+        Elements: (3 tokens) - (3 tokens) - (5 tokens) - (8 tokens)
+                    |            |            |            |
+                    (4 token summary) - (3 token summary) - (2 token summary)
+                                \\             |            /
+                                        (5 token summary)
 
+    Args:
+        llm: LLM to use for summarization
+        llm_mode: How to call the LLM - SYNC, ASYNC, BATCH. Async is faster but not all llms support it.
+        question: Optional question to use as context for the summarization. If set, the llm will
+            attempt to answer the question with the data provided
+        data_description: Optional string describing the input documents.
+        prompt: Prompt to use for each summarization. Caution: The default (MaxTokensHeirarchicalSummarizerPrompt)
+            has some fairly complicated logic encoded in it to make the tree construction work correctly.
+        fields: List of fields to include in each element's representation in the prompt. Specify
+            with dotted notation (e.g. properties.title), or use "*" to capture everything. If None,
+            will include no fields.
+        max_tokens: token limit for each summarization. Default is 10k (default tokenizer is by character).
+        tokenizer: tokenizer to use when computing how many tokens a prompt will take. Default is
+            CharacterTokenizer
+        max_rounds: max number of rounds of heirarchical summarization to perform. The number of elements that
+            can be included in the summary is O(e^max_rounds), so can be small. Default is 4.
+    """
 
-class QuestionAnsweringSummarizer:
-    def __init__(self, llm: LLM, question: str):
+    def __init__(
```

Comment:
Take a look at https://github.com/aryn-ai/sycamore/commit/03f5766e0fd4680ec1b151be2019780344646639
That's roughly the structure I was expecting.  I think the fake documents makes it easier to reason about as it reduces the layers of indirection.